### PR TITLE
feat: implement error handling middleware

### PR DESCRIPTION
**Pull request by arielsepton**
This pull request introduces an error handling middleware and custom error types to improve error management.

The `ErrorHandlingMiddleware` handles errors by intercepting any errors that occur during request processing. It extracts and categorizes errors into custom error types or Kubernetes status errors, then formats and sends a JSON response with the appropriate status code and error message.

The custom error types include:
* ErrorWithStatusCode: An interface for errors that include a status code.
* APIError: A generic error type with a message, status code, and reason.
* `ValidationError`, `NotFoundError`, `UnauthorizedError`, `ConflictError`, and `InternalServerError`: Specific error types for different scenarios, each with an appropriate HTTP status code and reason.

### CODE REVIEW

**Comment by arielsepton** [comment]
I still need to address the `c.Error` lint issue mentioned [here](https://github.com/gin-gonic/gin/issues/2017).

File: src/routes/context.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+func AddErrorToContext(c *gin.Context, err error) bool {
+	if err != nil {
+		// Add the error to the Gin context for further handling. Ignoring the return value as no action is needed based on it.
+		_ = c.Error(err)
+		return true
+	}
+	return false
+}
+
+// TODO: implement AddNewErrorToContext
```
Comment (dvirgilad) [inline src/routes/context.go:49]: Can this be removed?

**Review by dvirgilad — CHANGES_REQUESTED** [review]
Wow! Amazing work.
Left a few (minor) comments,

File: src/controllers/logs.go [hunk 2]
```diff
@@ -12,13 +13,20 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+const (
+	errCouldNotOpenLogStream = "error opening log stream"
+	errFetchingCappPods      = "error fetching Capp pods"
+	errNoPodsFound           = "no pods found for Capp %q in namespace %q"
+	errPodNotFound           = "pod %q not found for Capp %q in namespace %q"
```
[the remark this answers is not in the document]
Comment (arielsepton) [reply src/controllers/logs.go:20]: There is a difference between `no pods found` - meaning there are no pods and `Could not get pods` - indicating an error occurred in the fetching process.

File: src/controllers/containers.go [hunk 3]
```diff
@@ -3,13 +3,17 @@ package controllers
...
+	"github.com/dana-team/platform-backend/src/customerrors"
 	"github.com/dana-team/platform-backend/src/types"
-	"github.com/dana-team/platform-backend/src/utils"
 	"go.uber.org/zap"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
+const (
+	ErrCouldNotGetPod = "failed to get pod %q in the namespace %q"
```
[the remark this answers is not in the document]
Comment (arielsepton) [reply src/controllers/containers.go:14]: Despite the other errors that are thrown in pods.go, this error is thrown here.

File: src/controllers/capp.go [hunk 4]
```diff
@@ -28,6 +28,16 @@ const (
 	dnsLimit      = 10
 )
 
+const (
+	ErrCouldNotGetCapps     = "Could not get capps"
```
Comment (dvirgilad) [inline src/controllers/capp.go:32]: list

**Review by dvirgilad — APPROVED** [review]
/lgtm

**Pull request merged**